### PR TITLE
build: usage of dotenv for env variables

### PR DESCRIPTION
**Pull request by sshanzel**
Jira Ticket: [DD-203#done](https://dailydotdev.atlassian.net/jira/software/projects/DD/boards/1?selectedIssue=DD-203)

Packages added:
- dotenv: to manage environment variables. I have also applied a way to customize which dotenv file to be used. This is so env variables hidden from the repository can still be included in an ignored file.
- cross-env: even though most of us (if not all) are using macOS, I think it is still better to support different platforms.

It can be noticed that some `secret` and `salt` values are not excluded in the current `.env` since some of our repos already have those values which we will have to change, if needed, in the future at the same time.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
```
Comment (idoshamun) [inline .env:1]: You can remove this line

File: .env [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+CLOUDINARY_URL='cloudinary://848477452211613:u_2o8BUq_0CXdBOr_Ly-WcXtnAE@daily-now'
+
+WEBAPP_ORIGIN='http://localhost:5002'
+
+REFRESH_TOKEN_SECRET=topsecret
+REFRESH_TOKEN_SALT=salt
+
+AMPLITUDE_KEY=
+
+FLAGSMITH_KEY=
```
Comment (idoshamun) [inline .env:37]: Please add empty line here

File: src/config.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
+const defaultDotEnv = '.env';
+const customDotenv = process.env.APP_ENV && `.env.${process.env.APP_ENV}`;
```
Comment (idoshamun) [inline src/config.js:2]: I think it's better to use here the classic `NODE_ENV` env var. Otherwise, they might be redundant

**Review by idoshamun — CHANGES_REQUESTED** [review]
A few small changes but overall that's great.
I love that you decided to blend in cross env for Windows compatibility

File: src/config.js (hunk 3 above)
Comment (sshanzel) [reply src/config.js:2]: I was thinking if conflict might arise on how cloud infrastructures define `NODE_ENV`. Though as it seems might not be the case, let's go for the `NODE_ENV`.

**Review by idoshamun — APPROVED** [review]

**Pull request merged**